Dispatcher/UI/Search/SearchScreenViewController.swift

```diff
@@ -0,0 +1,115 @@
+        searchTableView.register(UINib(nibName: AppConstants.latestSearchCellNibName, bundle: nil), forCellReuseIdentifier: AppConstants.latestSearchesCellIdentifier)
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        searchScreenViewModel.fetchLatestSearchs()
+        searchTableView.reloadData()
+    }
+    
+    private func customSearchBar() {
+        let searchTextField = searchBar.searchTextField
+        searchBar.setImage(UIImage(named: "exit"), for: .clear, state: .normal)
+        
+        searchTextField.leftView = nil
+        searchTextField.backgroundColor = .white
+
+    }
+    
+    
+    @IBAction func backButtonPressed(_ sender: UIButton) {
+        self.dismiss(animated: true)
+    }
+    
+    @IBAction func clearButtonPressed(_ sender: UIButton) {
+        searchScreenViewModel.removeAllSearches()
+        searchTableView.reloadData()
+    }
+    
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        if (segue.identifier == AppConstants.fromLatestSearchToResults) {
+            let searchResultsVC = segue.destination as? SearchResultsScreenViewController
+            searchResultsVC?.searchKeyWords = searchBar.text ?? ""
+            searchResultsVC?.searchViewModel = searchScreenViewModel
+        }
+    }
+}
+
+
+extension SearchScreenViewController: UITableViewDataSource {
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return searchScreenViewModel.latestSearches.count
+        
+    }
+
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let currentSearchIndex = (searchScreenViewModel.latestSearches.count - indexPath.row) - 1
+        let search = searchScreenViewModel.latestSearches[currentSearchIndex]
+        
+        if let cell = (tableView.dequeueReusableCell(withIdentifier: AppConstants.latestSearchesCellIdentifier, for: indexPath) as? LatestSearchCell) {
+            
+            cell.searchWordsLabel.text = search
+            cell.removeButton.isHidden = false
+            cell.selectionStyle = .none
+            
+            cell.delegate = self
+            return cell
+      
+    }
+        
+        return LatestSearchCell()
+    }
+    
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        if let selectedCell = tableView.cellForRow(at: indexPath) as? LatestSearchCell {
+            searchBar.text = selectedCell.searchWordsLabel.text
+            performSegue(withIdentifier: AppConstants.fromLatestSearchToResults, sender: self)
+        }
+    }
+}
+
+extension SearchScreenViewController: UITableViewDelegate, UISearchBarDelegate, LatestSearchCellDelegate {
+    func searchCellSelected(_ search: String) {
+        searchBar.text = search
+        performSegue(withIdentifier: AppConstants.fromLatestSearchToResults, sender: self)
+    }
+    
+    func removeButtonPressed(_ searchToRemove: String) {
```

Comment:
We need a better name because when implementing the func here it's not clean enough what it's related to. Something like "removeSearchCellBtnPressed"